### DISCUSSION TITLE
Network graph is very misleading.

### DISCUSSION THREAD

**Discussion by fresnoboy** [zwave-js/zwave-js-ui#1420] [General]
HI. I just converted a zwave network from Homeseer to ZwaveJS, and the network graph was very misleading.  It indicated that battery powered locks were acting as routers, and even node neighbor info in there was wrong.  For example, a dome valve controller in graph says it's only neighbor is a lock (when I mouseover it), but it's still pingable and operates fine.

I know there is work going on to fix the graph, but this kind of misleading info had me chasing my tail for days and adjusting a perfectly functional network setup.  From threads on the home assistant boards, it appears this is a common occurrence.

Can the network graph function be removed until it presents reliable info?

**Comment by robertsLando** [comment]
> Can the network graph function be removed until it presents reliable info?

I see no reason why we should remove it, not all feedbacks about it are so bad

**Reply by sstarcher** [reply]
I have ran healing on my network a few times and targeted a few devices specifically.  
* I have a device 10 ft from my controller
* In the network graph it shows it going to a device 2 ft from my controller and it shows it going to a device 30 ft from my controller on a second story.

If that is actually how it's routing that seems very very bad.

**Reply by fresnoboy** [reply]
What version are you running?  Is that device that's 30 ft away a battery powered device?  Yes, the routing info displayed is often obviously wrong.

**Reply by sstarcher** [reply]
wall switch wired 10 ft away and 30 ft away is a light bulb so also wired.

zwavejs2mqtt 6.1.0
zwavejs 8.9.0-beta

**Reply by AlCalzone** [reply]
500 or 700 series stick?

**Reply by sstarcher** [reply]
700 series stick that I just updated to the latest 2.16 firmware.

**Reply by sstarcher** [reply]
ZST10-700

**Reply by fresnoboy** [reply]
I have the opposite problem in the current version.  All devices show they are directly connected to the Controller.  This is impossible because of the topology, and if I remove a repeater, many devices go offline.  I have a 700 series stick as well.

**Reply by AlCalzone** [reply]
The reason I asked is that the 700 series firmwares seem to do some really weird shit. It usually isn't noticeable for small networks, but this seems to be in line with what we're seeing.

**Reply by sstarcher** [reply]
This is my first zwave controller and I have certainly had tons of issues with it.   My network is about 40 devices I believe.

**Reply by AlCalzone** [reply]
> My network is about 40 devices I believe.

So on the larger-ish side. We're working with Silabs to get this resolved, but it will take a while.

**Reply by sstarcher** [reply]
Sounds great let me know if I can do anything to assist.

**Reply by AlCalzone** [reply]
Right now the ball is on Silabs' side again. They have a bunch of logs they'll have to work through and understand it.

**Reply by AlCalzone** [reply]
=> https://github.com/zwave-js/node-zwave-js/issues/3906

**Comment by fresnoboy** [comment]
It doesn't give accurate data about routing or even what neighbors are actually present.  I am happy to give you examples of this.  If it gives wrong data about those two critical sets of info, then it has much bigger opportunity for misleading people on faults, and increases mean time to repair.

If you believe it's routing info is accurate, or the neighbor data is accurate, I am happy to send you examples of where it isn't, though from the various forums, people seem to acknowledge it is wrong in both of those aspects.  But if you disagree with that I am happy to send the data.

**Comment by darkbasic** [comment]
I don't know about battery powered devices, but I think the network graph is great. Can you please share some examples of why do you believe it's bad? I personally would love to have SNR information for neighbours: the network graph should show information about the best SNR available for direct paths, 1 hop paths, 2 hop etc. I have some nodes with a terrible direct path SNR which fakes you into believing they are in direct range, while in reality most of the times they operate in 1 hope paths. That would be a really nice addition.
I also would love to have a subtitle showing the location under the node name, but apart from that I think it's great!

**Reply by msm8811** [reply]
The graph is very pretty.  It's also totally disconnected from what is actually happening in the network.   For example, mine regularly shows battery powered yale locks as routing nodes, which they can't be.  The neighbor info is also wrong.  I see nodes that only show no neighbors, but they can be communicated with, so that is also wrong.  There is almost nothing correct in the graph, at least in my setup.

The SNR data would be nice, but much more important is getting the neighbor and routing data to be presented correctly.  I had given examples in this ticket at the start, but you can look at the forums and see many many complaints on this, and even some of the core HA devs agreeing the network graph is incorrect.

The core zwavejs code doesn't seem to get this data right either, but displaying wrong data is far worse than not having a graph.

**Reply by darkbasic** [reply]
Are you saying that the neighbors information from zwave-js is plain wrong? Do you have any data to back it up?

**Reply by robertsLando** [reply]
ATM what I show in that graph is what comes from zwavejs. If data is not correct means there is something wrong on zwave-js side or maybe the controller has some problem/bug. Only @AlCalzone can say more about this

**Reply by darkbasic** [reply]
I will start adding my battery powered devices in a couple of days, I'll let you know if I notice any issue with the neighbors, because so far it seemed pretty accurate. On the contrary for the battery powered devices I will rely heavily on the network graph because most  of them will be out of direct reach.

**Comment by fresnoboy** [comment]
I am the OP.  Here is a nice pic to show you some of the issues:
![gragh](https://user-images.githubusercontent.com/52867314/136067538-1a2a94a5-bbc7-4cdc-889b-d94b05c79ac9.jpg)


Note the battery powered poolhouse door and side door yale locks are shown as routers, which they can't be.  The rec room lock says it has no neighbors when I  mouse over it, though it still works, yet obviously has a route.  

It may be that zwavejs has good data and the graph is not displaying it properly, but the data represented in the graph is very often wrong.

See these threads for supplemental info:

https://community.home-assistant.io/t/network-graph-shows-battery-device-as-controllers-only-neighbor/338337/3

https://community.home-assistant.io/t/z-wave-graph-without-the-python/64275/461

https://community.home-assistant.io/t/zwavejs2mqtt-ui-seems-disconnected/297741

https://community.home-assistant.io/t/zwavejs-routing-through-battery-powered-locks/321402/20

https://community.home-assistant.io/t/70-zwave-device-network-stuck-healing-for-over-24hrs/332425/23

There are others, but you can get the general feel for widespread issues with the network graph.

**Reply by darkbasic** [reply]
> Note the battery powered poolhouse door and side door yale locks are shown as routers, which they can't be.

That's weird, but as I said I still have to include my battery powered devices.

> The rec room lock says it has no neighbors when I mouse over it, though it still works, yet obviously has a route.

I see a connection to the Range Extender, so it obviously must have a neighbor and it should show it when you hover over it. Not sure why it doesn't.

**Reply by fresnoboy** [reply]
Of course battery powered devices should be included in the graph, but they can't be routers, so why is the code showing they are?

I assure you the rec room lock says it has no neighbors, but it does work, and I don't know why you show it routed the way it is when you are saying it has no neighbors.  

These are common issues reported in the threads I referenced.  I can find more examples of users complaining and confused by the graph reports if you like.

**Reply by darkbasic** [reply]
> Of course battery powered devices should be included in the graph

I just meant that I still have to include them **in my network** (which I'm rebuilding from scratch) and thus I have no experience with battery powered devices in the graph.

> I assure you the rec room lock says it has no neighbors

I believe you, I'm just saying that the graph shows a link with another node.

**Reply by robertsLando** [reply]
@fresnoboy What version are you using?

**Reply by fresnoboy** [reply]
zwavejs2mqtt: 5.6.0
zwave-js: 8.3.1

**Reply by robertsLando** [reply]
Could you give a try to https://github.com/zwave-js/zwavejs2mqtt/releases/tag/v5.6.1

or a newer version? There has been a fix in the forwarding checks for battery devices

**Reply by AlCalzone** [reply]
> It may be that zwavejs has good data and the graph is not displaying it properly, but the data represented in the graph is very often wrong.

Aside from the issue that battery powered nodes should not be displayed as routing, I can assure you, the info that zwave-js has access to is also wrong very often.

People seem to focus too hard on this graph IMO. It doesn't tell you much, except which node sees which neighbors, as long as the controller has this information already (which it might not).

It does not tell you which routes are being used, it does not tell you how well the nodes see each other. And sometimes, routes are actually being used that seem impossible by looking at the graph.

**Reply by fresnoboy** [reply]
Al, that is consistent with many comments on the forums.  ZwaveJS is materially much worse in terms of any real info about the Zwave Mesh than is available from Homeseer or even Hubitat.  I came from Homeseer, and it told me the actual routing and data rates of each link, and even allowed me to manually specify a return route (I think routes in Zwave do not have to be symmetrical).  Both of these companies have relationships with silicon labs, so not clear if they have special access to drivers etc...   

I think I finally have my configuration stable (it was stable when I converted from Homeseer, but I started breaking things when I saw the network graph showing routing that was much different than before), else I would probably have migrated everything to my hubitat Zwave radio.  All this bad data is why the graph is so misleading and causes people to waste time and break things trying to fix routing and bad connectivity.  Better to have no data than wrong data.  

Oh, I did upgrade to the latest version.  Now the rec room lock is showing 1 neighbor, but the graph still shows routing through battery powered locks which is impossible:


![graph2](https://user-images.githubusercontent.com/52867314/136219653-b4ddb368-40b8-4b23-94ad-61dfa64b517b.jpg)

**Reply by darkbasic** [reply]
> it told me the actual routing and data rates of each link, and even allowed me to manually specify a return route

That would be highly desirable of course, but the current implementation is still better than nothing IMO (minus the bugs, which must be fixed).

**Comment by fresnoboy** [comment]
PS I often do check version on the zwavejs2mqtt instance that runs on a raspberry pi.  It didn't tell me that there was a new version, or at least if it did I missed it.  When I initiate the check, what is supposed to happen if an upgrade is available?

**Reply by robertsLando** [reply]
The new version is just for config files updates, not for zwavejs2mqtt updates

**Reply by fresnoboy** [reply]
Ah, I get it.  Thanks.

 Is HA supposed to prompt you that there is a zwave2mqtt update available?  How are users supposed to know an update to the docker is available?

**Reply by robertsLando** [reply]
@frenck is the hass addon maintainer and maybe could answer to this, I have no clue

**Reply by kpine** [reply]
> Is HA supposed to prompt you that there is a zwave2mqtt update available? How are users supposed to know an update to the docker is available?

If you are using the Community Addon, then the Supervisor will notify you, or might even be able to auto-upgrade. It sounds like you are using a standalone Docker installation? In that case, since this project has no relation to the HA project, there's no reason HA would notify you of a new release.

There are probably a ton of ways to auto-update containers or get notifications, but see [Watchtower](https://containrrr.dev/watchtower/) or [Diun](https://crazymax.dev/diun/) for some popular examples. I personally use Github notifications. You can watch releases for this project and get an email anytime a release is published. The docker image will be available either before the email is sent, or maybe soon after. When I get an email I check the release notes, and update the image and re-create the container if I actually need the update.

**Reply by fresnoboy** [reply]
Thanks for the reply.

Yes, I run zwavejs2mqtt on a raspberry pi located in the central part of the house closest to the devices, as opposed to my HA instance, which runs as a guest VM on a computer in the basement.  This allows me to not to need USB passthrough etc... on the VM.  Same with deconz driving zigbee.  

Ok, so sounds like there is no notification of updates in the zwavejs2mqtt UI, or a flag via websockets to drive a HA notification?  I'll try the github approach.

**Comment by fresnoboy** [comment]
BTW, attached is the help file from Homeseer's zwave controller utility.  This is for HS3 and a bit dated, but you can see the functionality that is built into that system for Zwave support.

What version of the Silicon Labs zwave library is the current version of the zwavejs using?  I think Homeseer is now using 7.14, since it fully supports the features of the 700 series stick.

[Z-Wave_Help_File.pdf](https://github.com/zwave-js/zwavejs2mqtt/files/7295889/Z-Wave_Help_File.pdf)

**Reply by AlCalzone** [reply]
Z-Wave JS isn't built against a certain SDK. It implements and manages the controller commands on its own, based on the Serial API documentation. Part of the reason why certain commands aren't implemented yet, but this way no native libraries need to be compiled, aside from `node-serialport`.

Can you point me to anything special in the PDF? Would be interesting to see how certain things are implemented elsewhere, but on the first glance I didn't spot anything other than "click this, then that"

**Reply by fresnoboy** [reply]
If you look at the commands, it can tell you the actual route being used, and also supports scene management, which is very handy if you are using zwave remotes, etc...  without resorting to using parameters and bitmasks, etc...  Homeseer understands what all those parameters mean and maps them to cross device capabilities that are a lot easier to configure.  It also tells you speeds of each link, so you can tell if the signal has quality problems, and you can also set the return route used manually, which is also helpful for troubleshooting the mesh.  It also has a set of optimization commands that you can use to help reconfigure and optimize the mesh after changes are made, as well as utilities for backing up controllers and restoring them, which is important for the 700 series controllers, because the internal data formats of 500 and 700 series are incompatible.  With homeseer, you can convert the network stored in the 500 stick to the 700 stick format and upgrade the controller.  

Also, if you look at their zseer test app that is  used for debugging: https://homeseer.com/pdfs/guides/Z-Seer_Installation_Guide.pdf, you can see how they generate a map with useful data that is correct and not wrong.

Based on my read of the Silicon Labs documentation, and experience with other controllers, it's very clear to me that the serial API is inadequate, and you have to use their API if you want full zwave functionality.  This is the key to simplifying configuration and actually understanding what is going on with the mesh.  Given ZwaveJS is now the primary Zwave interface for HA, it would seem that Silicon Labs would want you to have access to their library or at least the library code that interacts with their serial API to enable full functionality for HA users.  Because without that information, ZwaveJS will ALWAYS be inferior to Home Automation systems that do use it (in terms of Zwave functionality), and HA users will tend to move to Zigbee, WiFi  and alternate protocols that do have full functionality supported.   That is not in Silicon Labs best interest, because they make money from people buying their Zwave chips.   So they should want to work with you!

I am a former Homeseer user, and HA is superior in almost every way, except for it's Zwave support, which is why I am hoping that you guys can get to parity and close that gap.

**Reply by AlCalzone** [reply]
I believe there is at least some kind of misunderstanding involved on your part. The official controller SDK is "just" another wrapper around the (documented) Serial API. Switching to it would be a complete rewrite of Z-Wave JS at this point, so this is off the table.

Some of what you mention is a limitation of HomeAssistant's integration and is already supported by the driver (`node-zwave-js`) and to a slightly lesser extent zwavejs2mqtt. Examples are:
* scene management (https://github.com/zwave-js/node-zwave-js/pull/3377, https://github.com/zwave-js/node-zwave-js/pull/3354, https://github.com/zwave-js/node-zwave-js/pull/3325, https://github.com/zwave-js/node-zwave-js/pull/3313)
* utilities for backing up controllers and restoring them (500 series supported, 700 series [issue](https://github.com/zwave-js/node-zwave-js/issues/3482))
* resorting to using parameters and bitmasks - zwavejs2mqtt offers simple dropdowns for each config parameter with a proper label. HA is working on it (might be included in the october release, not sure)
![grafik](https://user-images.githubusercontent.com/17641229/136707294-fc88cb44-6069-4d45-a7b8-2223ad15a160.png)

Other commands need to be implemented in the driver, but this takes time. I've had some help, but most of this core work is still a one-man operation.

> you can see how they generate a map with useful data that is correct and not wrong.

They do have better utilties for testing the network quality, but at least the screenshot looks like the neighbor information is the same that Z-Wave JS also requests from the controller. Like I said, this is often wrong, but this either means that the controller has a wrong topology (I have no clue what users do with their networks) or it isn't up to date. I've already seen logs where the controller claims it has no neighbors - not sure how you would even achieve that 🤷🏻‍♂️

> the internal data formats of 500 and 700 series are incompatible. With homeseer, you can convert the network stored in the 500 stick to the 700 stick format and upgrade the controller.

Oh, I've dabbled with that already. The 500 series memory formats are a huge fucking mess, change between minor/patch SDK versions and even the information I got my hands on isn't enough to make sense of them (e.g. a 16 bit number to address potential 2^24 bits of memory). So, HomeSeer must have a very good contact at Silabs to help them through it.

**Reply by AlCalzone** [reply]
FWIW, this is my little network which I built over the last couple of years:
![grafik](https://user-images.githubusercontent.com/17641229/136708069-185f886c-645b-4509-a99e-ade85d1f62c5.png)

**Reply by fresnoboy** [reply]
> I believe there is at least some kind of misunderstanding involved on your part. The official controller SDK is "just" another wrapper around the (documented) Serial API. Switching to it would be a complete rewrite of Z-Wave JS at this point, so this is off the table.
> 
> Some of what you mention is a limitation of HomeAssistant's integration and is already supported by the driver (`node-zwave-js`) and to a slightly lesser extent zwavejs2mqtt. Examples are:
> 
> * scene management ([fix(cc): add prepopulate `Scene Actuator Configuration CC` `valueMetaData` node-zwave-js#3377](https://github.com/zwave-js/node-zwave-js/pull/3377), [feat(cc): add `compat` for `Scene Controller Configuration CC` node-zwave-js#3354](https://github.com/zwave-js/node-zwave-js/pull/3354), [feat(cc): add setting `dimmingDuration` for `Scene Actuator Configuration CC` with `setValue` API node-zwave-js#3325](https://github.com/zwave-js/node-zwave-js/pull/3325), [feat(cc): add setting `dimmingDuration` for `Scene Controller Configuration CC` with `setValue` API node-zwave-js#3313](https://github.com/zwave-js/node-zwave-js/pull/3313))
> * utilities for backing up controllers and restoring them (500 series supported, 700 series [issue](https://github.com/zwave-js/node-zwave-js/issues/3482))
> * resorting to using parameters and bitmasks - zwavejs2mqtt offers simple dropdowns for each config parameter with a proper label. HA is working on it (might be included in the october release, not sure)
>   ![grafik](https://user-images.githubusercontent.com/17641229/136707294-fc88cb44-6069-4d45-a7b8-2223ad15a160.png)
> 
> Other commands need to be implemented in the driver, but this takes time. I've had some help, but most of this core work is still a one-man operation.
> 
> > you can see how they generate a map with useful data that is correct and not wrong.
> 
> They do have better utilties for testing the network quality, but at least the screenshot looks like the neighbor information is the same that Z-Wave JS also requests from the controller. Like I said, this is often wrong, but this either means that the controller has a wrong topology (I have no clue what users do with their networks) or it isn't up to date. I've already seen logs where the controller claims it has no neighbors - not sure how you would even achieve that 🤷🏻‍♂️
> 
> > the internal data formats of 500 and 700 series are incompatible. With homeseer, you can convert the network stored in the 500 stick to the 700 stick format and upgrade the controller.
> 
> Oh, I've dabbled with that already. The 500 series memory formats are a huge fucking mess, change between minor/patch SDK versions and even the information I got my hands on isn't enough to make sense of them (e.g. a 16 bit number to address potential 2^24 bits of memory). So, HomeSeer must have a very good contact at Silabs to help them through it.

Thanks for the detailed reply.  On the topic of the node that ZwaveJS is saying has no neighbors but works, clearly that's not the case.  So either the node (a zwave lock) is reporting bad data, or ZwaveJS is not dealing with the network topology interaction properly.  I have never seen this happen under Homeseer, so I don't think it's an reporting error from the lock.  Note that one of the other nodes shows the lock as a neighbor, but ZwaveJS still says the lock has no neighbors, so there is a problem here.

Your example network is quite neat!  But many of us have larger areas covered when the Zwave controller must route through other nodes to reach remote devices.  You clearly don't have that situation if your graph is correct.  I believe ZwaveJS may display your topology correctly, but it's certainly not doing it for more "meshy" topologies.

I'm glad you have a set of fixes and new work under way to improve things.  My main point was that silicon labs should have strong incentives to help you make this work really well, and they may be able to reduce the workload making these improvements by sharing code examples or other data with you.  You really should be working closely with them, esp if this is mostly a one man project.  Whether that involves using their libraries, or getting their help, the incentives should be aligned for them to give you a lot of assistance. Clearly the other platforms that are using their libraries have at least better diagnostics and correct maps, and I doubt they are better programmers than you guys are!  So I surmise that Silicon Labs libraries and help is probably the difference.  You should be leveraging that help too!

**Reply by AlCalzone** [reply]
> On the topic of the node that ZwaveJS is saying has no neighbors but works, clearly that's not the case. 

Yesterday we stubled across something in the 7.16 SDK release notes. Apparently the previous 700 series firmwares had a too high TX power while including nodes, which may result in nodes being included directly when they should be included with at least 1 hop in between.

I wonder if all (or most of) these borked network maps are on 700 series sticks and are the result of this firmware bug. At least it could explain it: Nodes are included as "direct range" due to too high TX power. Later they don't see the controller anymore, leading to missing links or pseudo routes through battery nodes when it should have been the other way round.

Maybe you can help support our thesis:
* Which controller did you migrate to from HomeSeer? 500 or 700 series?
* How far away are the nodes with incorrect routes from the controller?

**Reply by darkbasic** [reply]
@AlCalzone damn I have some nodes which don't support NWI which barely included with 7.11, are you saying that I won't be able to re-include them with 7.16.2 now? Any way to revert to max TX power during inclusion?
Also isn't this something which would be easily fixed by a simple heal?

**Reply by AlCalzone** [reply]
If they barely included on 7.11, the inclusion probably shouldn't have worked at all because the reduced power makes sure you have enough of a signal to noise tolerance during usage later, and therefore a stable mesh.

Increasing TX power, i.e. shouting louder doesn't help if you can't also "listen better". Also I'm not sure if there is a way to do that during inclusion.

**Reply by darkbasic** [reply]
> Increasing TX power, i.e. shouting louder doesn't help if you can't also "listen better"

It does because I can run the inclusion multiple times until it succeeds and then I won't have any communication issue because it will route packets through the mesh, while it cannot do so during inclusion because of the lack of NWI.

**Reply by AlCalzone** [reply]
You're right, I didn't consider the lack of NWI. You'll have to include the node in direct range anyways and then fix the mesh. The driver offers methods to [control the TX power](https://zwave-js.github.io/node-zwave-js/#/api/controller?id=configure-tx-powerlevel). I'm not sure though what the effect on inclusion will be.

**Reply by fresnoboy** [reply]
I have a 700 stick, but the conversion from the 500 stick to the 700 stick was done before my switchover to ZwaveJS2MQTT.  And the node that is reporting the lock as a neighbor is not the controller, but a aeotec zwave repeater.

Agree that TX power is not that helpful for range, unless you have a 700 series repeater with the same TX power levels.  But it does help with normal devices handling faster data rates from the controller.  

The repeater in my network that reports the lock as a neighbor is visible to the controller, but the lock is not.  In general devices are scattered reasonably far since I use Lutron for lighting control, and zigbee for almost anything else.  However, other locks that ZwaveJS says are router nodes are close to the controller.

**Reply by AlCalzone** [reply]
> but the conversion from the 500 stick to the 700 stick

Did you do anything special or just excluded the nodes on one stick and re-included on the other? Because the mentioned issue isn't specific to Z-Wave JS, but rather the 700 series stick firmware.

**Reply by fresnoboy** [reply]
Sorry if I wasn't clear.  I converted from the 500 stick to the 700 stick (running 7.15) in homeseer.  I then exported the security key, and brought the network into ZwaveJS2MQTT without any re-pairing.  Everything seemed to work, but when I saw the map showing crazy routing and neighbors out of whack, I immediately started troubleshooting and removing and reincluding devices and moving things around.  This is what actually caused a ton of instability and brokenness.

So when I reached out about this on the HA forums, people told me the network graph was all wrong and not to believe what it said.  Once I knew the data could not be trusted, I did a bunch of trial and error and got the network stable again over the course of a month.  When I asked why such a misleading set of info was being reported, one of the mods said that HA wasn't responsible for this code and I should start a discussion here.  :)

**Reply by AlCalzone** [reply]
Do you still have access to HomeSeer and can check if it reports the same neighbors?

**Comment by kpine** [comment]
Regarding the comments about battery devices being shown as "routers" in the default network graph view... [This change](https://github.com/zwave-js/zwavejs2mqtt/commit/e5cf1515903adf3d4c6374e86efebe5509d0bf26) modified the graph to only show devices which report `isRouting` as forwarding nodes. Despite that, my battery-powered smoke alarms and door sensors still show up as routers/forwarding nodes, and I believe this is because they are certified as "Routing Slaves", so the `isRouting` property is true. I don't know why they would be considering routing slaves as non-FLIRS battery devices?

Based on the driver documentation for [`isRouting`](https://zwave-js.github.io/node-zwave-js/#/api/node?id=isrouting):

> If **both** this property and isListening are true, it can be assumed that the node will route messages.

I'm wondering if the check should be this instead:
```javascript
forwards:
  node.isControllerNode ||
  (node.ready && !node.failed && node.isRouting && node.isListening),
```

Should it be assumed that they will _not_ route messages if `isRouting` is true but `isListening` is false?

Patio Door is a door sensor, Node 11 is a smoke alarm.
![image](https://user-images.githubusercontent.com/791794/136709043-0646b49e-b7d6-4109-95e9-330bcf545b00.png)

Maybe the default view should be "All Edges" so there's no confusion that the network graph is **not** a routing graph. There's no mistaking that a node is a router in this view. I have no clue what the line colors and fill mean though.

![image](https://user-images.githubusercontent.com/791794/136709087-5667ce54-041e-4ef9-99e6-678c7d21ebce.png)

**Reply by kpine** [reply]
I think every non-controller battery device in existence is either "Routing slave" or "Enhanced 232 Slave" library type (not sure if that's what's parsed in the NIF). All the battery devices in my network, except for a portable controller, report true for isRouting, to that is obviously not the correct criteria for the graph.

**Reply by AlCalzone** [reply]
I discussed this with @robertsLando about 3 weeks ago. It seems like the discussed change didn't make it into the codebase correctly:
https://github.com/zwave-js/zwavejs2mqtt/commit/e5cf1515903adf3d4c6374e86efebe5509d0bf26

**Reply by robertsLando** [reply]
Fixed on master now
